kibana.md

```diff
@@ -1101,3 +1101,189 @@ an admin, who has just configured access for a new user. They would like
+| getMethod()                                                            | Method                       | Get the request method                                                         |
+| getPath()                                                              | string                       | Get the request path                                                           |
+| getUrl()                                                               | string                       | Get the request URL                                                            |
+| getBody()                                                              | Body or null or undefined    | Get the request body                                                           |
+| getParams()                                                            | ParamsDictionary             | Get URL parameters                                                             |
+| getQueries()                                                           | ParsedQs                     | Get query string parameters                                                    |
+| getOriginAddress()                                                     | string undefined             | Get the origin address                                                         |
+| getHeaders()                                                           | Headers                      | Get all request headers                                                        |
+| isAuthenticated()                                                      | boolean                      | Check if the session is authenticated                                          |
+| isCookiePresent(cookieName: string)                                    | boolean or undefined         | Check if the specific cookie is presented in the request                       |
+| getIdentitySession()                                                   | IdentitySession or undefined | Get the session identity (check the information below, for the exact response) |
+| setIdentitySession(identitySession: IdentitySession or undefined)      | void                         | Set the new session                                                            |
+| enrichIdentitySessionMetadata(customMetadata: Record<string, unknown>) | void                         | Enrich existing user session by the additional custom metadata                 |
+| lastSessionActivityDate                                                | Date or undefined            | Date of the last session activity.Using in the context of a session timeout    |
+| extractHiddenAppsNames                                                 | string[]                     | List of all hidden apps for specific users |
+
+```ts
+export interface IdentitySession {
+  sid: string;
+  metadata: {
```

Comment:
wait a sec ... do we want to expose all of these? During the analysis, I argued about encapsulation and I was not a fan of exposing ROR internals to the user's code. So we decided to create a metadata field in the `kibana` rule (see [jira](https://readonlyrest.atlassian.net/jira/software/projects/RORDEV/boards/2?selectedIssue=RORDEV-837)). And only this data should be available to the user.

Reply:
Yeah, that's the plan, and we even expose all metadata for the `kibana_custom_js_inject_file` purpose. So the idea is to expose only custom metadata (I understand the metadata field this way). AFAIR it was a business decision to expose everything to the user WDYT @sscarduzio ?

Reply:
But keeping anything as a public interface is an effort. It's better to have not so many public interfaces. I'm an advocate of exposing as little as possible. Do we know how users would use information about `kibana hidden` apps? or `kibana access`? or impersonation ... or a such internal thing like `correlationId`. 

Currently, I'm working on supporting vars usage in ROR ES `kibana.metadata` value. If we have that why do we need to expose `availableGroups`, `currentGroup`, `username` in the identity object? 

IMO we should expose as little as possible. Narrow the public interface. When we see how users use the feature and know the limitations (from their feedback) we could discuss how to expose more in the safest way. It's very hard to fix public interfaces in a seamless way for the users.

Reply:
NB: these properties are not exposing the internal messaging format between ES and Kibana, rather they are exposing the configuration parameters of the Kibana+ROR UI. 

* `availableGroups`, `currentGroup`, `username`, `kibana_access`, `hidden_apps` all define the Kibana UI state and front end developers need those available to extend ROR with custom behaviours, or they are going to pick this info out of the DOM, which is REALLY unstable.

* correlation_id is not really internal, is it? it was added for audit IIRC? BTW customMiddleware is the perfect point where to generate custom audit data points, we should expose an API for that (in phase2 of this)

Reply:
> If we have that why do we need to expose availableGroups, currentGroup, username in the identity object?

For string composition/manipulation on the ACL side mainly

Reply:
> > If we have that why do we need to expose availableGroups, currentGroup, username in the identity object?
> 
> For string composition/manipulation on the ACL side mainly

Does it have a real value when you have the same in the context of JS code? You can manipulate the strings with no limitation

Reply:
As I said availableGroups, currentGroup, username, kibana_access, hidden_apps all describe the UI state, and they are concepts that are already perceived as "already known to Kibana" in a normal session (i.e. people see the tenancy selector, the username, etc in the UI all the time). 
Writing in the documentation "if you want to use these variables you have to tell Elasticsearch to send them" is a mental shortcircuit.  Plus, I really don't see these basic things changing anytime.